processing/src/main/java/io/druid/query/filter/SpatialDimFilter.java

```diff
@@ -111,14 +122,15 @@ public boolean equals(Object o)
       return false;
     }
 
-    return true;
+    return Objects.equal(delimiter, that.delimiter);
```

Comment:
Also this needlessly uses Guava's utility class and methods, which are deprecated since Java 7 in favor of java.utils.Objects.